packages/common/src/store/pages/chat/slice.ts

```diff
@@ -284,8 +296,22 @@ const slice = createSlice({
     ) => {
       // triggers saga to get chat if not exists
       const { chatId, message, status } = action.payload
+
+      // Recalculate hasTail of previous message
+      const prevLatestMessageId = state.messages[chatId].ids[0]
+      const prevLatestMessage =
+        state.messages[chatId].entities[prevLatestMessageId]
+      if (prevLatestMessage) {
+        const prevMsgHasTail = hasTail(prevLatestMessage, message)
+        chatMessagesAdapter.updateOne(state.messages[chatId], {
```

Comment:
nit: ignore if idk what im talking about, but maybe we should only update if `prevMsgHasTail` is diff than current value

Reply:
Great callout. I think it doesn't matter much if we update with the same value as before. There's a potential argument for slightly better performance if we do the check for if `prevMsgHasTail` is diff from current value, but I'd personally opt for one less if statement in the code. Lmk what you think @sddioulde

Reply:
the reason why I called it out is because there may be selectors that use this value that would have to update in the UI. but yeah, this is fine

Reply:
Ah good call. I think since we're in `addMessage` the UI is gonna update anyway so in this case doesn't matter, but I will remember to think of this.